lms/djangoapps/teams/static/teams/js/views/teams_tabbed_view.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * A custom TabbedView for Teams.
+ */
+;(function (define) {
+    'use strict';
+
+    define([
+        'js/components/tabbed/views/tabbed_view',
+        'teams/js/utils/team_analytics'
+    ], function (TabbedView, TeamAnalytics) {
+        var TeamsTabbedView = TabbedView.extend({
+            /**
+             * Overrides TabbedView.prototype.setActiveTab in order to
+             * log page viewed events.
+             */
+            setActiveTab: function (index) {
+                var pageName = (typeof index === 'string') ? this.urlMap[index].url : this.tabs[index].url;
```

Comment:
This is duplicating some logic in `TabbedView.setActiveTab`; you could factor it out into a function to get the tab information from a number or string.